src/platform/Tizen/ChipDeviceScanner.cpp

```diff
@@ -0,0 +1,321 @@
+    return true;
+}
+
+void ChipDeviceScanner::CheckScanFilter(ScanFilterType filterType, ScanFilterData & filterData)
+{
+    int ret = BT_ERROR_NONE;
+
+    // Scan Filter check
+    if (!__IsScanFilterSupported())
+        return;
+
+    ret = CreateLEScanFilter(filterType, filterData);
+    VerifyOrExit(ret == BT_ERROR_NONE, ChipLogError(DeviceLayer, "Scan Filter Creation Failed! ret: %d Do Normal Scan", ret));
+
+    ret = RegisterScanFilter(filterType, filterData);
+    VerifyOrExit(ret == BT_ERROR_NONE, ChipLogError(DeviceLayer, "Scan Filter Registration Failed! ret: %d Do Normal Scan", ret));
+    return;
+exit:
+    UnRegisterScanFilter();
+}
+
+CHIP_ERROR ChipDeviceScanner::StartChipScan(unsigned timeoutMs, ScanFilterType filterType, ScanFilterData & filterData)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    ReturnErrorCodeIf(mIsScanning, CHIP_ERROR_INCORRECT_STATE);
+
+    // Scan Filter Setup if supported: silently bypass error & do filterless scan in case of error
+    CheckScanFilter(filterType, filterData);
+
+    kScanTimeout = timeoutMs;
+
+    // All set to trigger LE Scan
+    ChipLogProgress(DeviceLayer, "Start CHIP Scan...");
+    if (MainLoop::Instance().AsyncRequest(TriggerScan, this) == false)
+    {
+        ChipLogError(DeviceLayer, "Failed to trigger Scan...");
+        err = CHIP_ERROR_INTERNAL;
+        goto exit;
+    }
+
+    mIsScanning = true; // optimistic, to allow all callbacks to check this
+    return CHIP_NO_ERROR;
+
+exit:
+    ChipLogError(DeviceLayer, "Start CHIP Scan could not succeed fully! Stop Scan...");
+    StopChipScan();
+    UnRegisterScanFilter();
+    return err;
+}
+
+CHIP_ERROR ChipDeviceScanner::StopChipScan(void)
+{
+    int ret = BT_ERROR_NONE;
+    ReturnErrorCodeIf(!mIsScanning, CHIP_ERROR_INCORRECT_STATE);
+
+    ret = bt_adapter_le_stop_scan();
+    ChipLogError(DeviceLayer, "bt_adapter_le_stop_scan() ret: %d", ret);
+
+    g_main_loop_quit(mAsyncLoop);
+    ChipLogProgress(DeviceLayer, "CHIP Scanner Async Thread Quit Done..Wait for Thread Windup...!");
+
+    UnRegisterScanFilter();
+
+    // Report to Impl class
+    mDelegate->OnChipScanComplete();
+
+    mIsScanning = false;
+
+    return CHIP_NO_ERROR;
+}
+
+void ChipDeviceScanner::UnRegisterScanFilter(void)
+{
+    if (mScanFilter)
+    {
+        bt_adapter_le_scan_filter_unregister(mScanFilter);
+        mScanFilter = nullptr;
+    }
+}
+
+int ChipDeviceScanner::RegisterScanFilter(ScanFilterType filterType, ScanFilterData & filterData)
+{
+    int ret = BT_ERROR_NONE;
+
+    switch (filterType)
+    {
+    case ScanFilterType::kAddress: {
+        ChipLogProgress(DeviceLayer, "Register Scan filter: Address");
+        ret = bt_adapter_le_scan_filter_set_device_address(mScanFilter, filterData.address);
+        VerifyOrExit(ret == BT_ERROR_NONE,
+                     ChipLogError(DeviceLayer, "bt_adapter_le_scan_filter_set_device_address() failed. ret: %d", ret));
+        break;
+    }
+    case ScanFilterType::kServiceUUID: {
```

Comment:
Be careful with using this filter. The spec doesn't require that Matter devices put standalone UUID fields in the advertising data. Only the service data field must contain the service UUID.

Reply:
@Damian-Nordic , Yes, agreed & noted!
However, for now, just leaving this additional case as it is, because, we are using *ServiceData* filter explicitly for BLE Scan. Thank you for your review~